### ISSUE TITLE
Allow control over path of extracted library

### ISSUE THREAD

**Issue by oferfort** [dmlc/xgboost#10302]
In the jvm-package, the NativeLibLoader extracts the library from the jar to the java.io.tmpdir. We prefer to be able to control that path for the xgboost library independently.
Thanks!

**Comment by trivialfis** [comment]
Hi, could you please share more details on your use case, and what type of paths would you use?

**Comment by oferfort** [comment]
We are running in an environment where the default tmp folder has limited permissions.
We would like to be able to control the folder the lib is extracted to, so that it can be managed, independently.
Regards

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by wbo4958** [comment]
probably change "java.io.tmpdir" accordingly? @oferfort

**Comment by oferfort** [comment]
There are other applications on the same JVM so we can't do that

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by wbo4958** [comment]
Yeah, it's quite doable, but recently I am working on https://github.com/dmlc/xgboost/issues/10415. Will do that if no one picks this issue.
